core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts

```diff
@@ -262,34 +157,32 @@ export class ServiceMetricsComponent implements OnInit, AfterViewInit {
         statistics: this.filters.getFieldValueOfLabel('AGGREGATION')
       }
     };
-    return this.http.post(request.url, request.body, this.service.id)
+    return this.http.post(request.url, request.body)
       .toPromise()
       .then((response) => {
         this.sectionStatus = 'empty';
-        if (response && response.data && response.data.assets && response.data.assets.length) {
+        if (response && response.data && response.data.length) {
           this.queryDataRaw = response.data;
-          this.queryDataRaw.assets = this.filterAssetType(response.data);
-          this.setAssetsFilter();
-          this.setAsset();
+          this.getAllData(response.data);
         }
-      },
-      (error) => {
+      })
+      .catch((error) => {
         this.sectionStatus = 'error';
-        this.errorData['response'] = error;
-      });
-
+        console.log(error);
+      })
+  }
+  getAllData(data) {
+    this.allData = data;
+    this.queryDataRaw.assets = this.filterAssetType(this.allData[0]);
+    this.setAssetsFilter();
+    this.setAsset();
   }
-
   filterAssetType(data) {
     return data.assets.filter((asset) => {
       if (this.serviceType === 'api') {
-        if (this.service.deployment_targets === "gcp_apigee") {
```

Comment:
Include this if-else conditions. (Looks like you need to take pull from upstream)

Reply:
Included this